### ISSUE TITLE
Docker 20.10.23 VSC 1.74.3 DEv Containers v0.266.1 Expose Ports Does not Work as expected

### ISSUE THREAD

**Issue by razilevin** [microsoft/vscode-remote-release#7877]
Seems ports do not get exposed by default (atleast in my case) when running the following without adding the app port property. Did not need that before my update. All the configuration is stock configuration generated by VSC when using the Node:18 remote container.

Not sure of root cause.

Dev Containers Version

`v0.266.1 `

VSC Code Release Information

```
Version: 1.74.3
Commit: 97dec172d3256f8ca4bfb2143f3f76b503ca0534
Date: 2023-01-09T16:57:40.428Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.15.0-58-generic
Sandboxed: No

```

`Docker version 20.10.23, build 7155243`

OS Information

```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy

```
devcontainer.json

```
// See https://containers.dev/implementors/json_reference/ for configuration reference
{
	"name": "Untitled Node.js project",
	"build": {
		"dockerfile": "Dockerfile"
	},
	"remoteUser": "node",
	"appPort": 3000
}

```

Dockerfile

```
FROM node:18

# Install basic development tools
RUN apt update && apt install -y less man-db sudo

# Ensure default `node` user has access to `sudo`
ARG USERNAME=node
RUN echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
    && chmod 0440 /etc/sudoers.d/$USERNAME

# Set `DEVCONTAINER` environment variable to help with orientation
ENV DEVCONTAINER=true

```

**Comment by razilevin** [comment]
Update: Seems app port just creates a statically forwarded port which did not fix the problem. I had to add the port again so you see this in the ports tab of the VSC

```
3000 127.0.0.1:3000 /usr/local/bin/node... User Forwarded
3000 127.0.0.1:3000 <blank>                         Statically Forwarded
```

Here is the output of process
```
node@0dd5593a3f88:/workspaces/dynamic-category-contentstack-preview$ npm run dev

> dynamic-category-contentstack-preview@1.0.0 dev
> fastify start -w -l info -P app.js

16:23:06 ✨ Server listening at http://0.0.0.0:3000
16:23:06 ✨ Server listening at http://0.0.0.0:3000
```

Only after adding the port manually I see this on the host machine

```
docker port  sharp_chebyshev
3000/tcp -> 127.0.0.1:3000
```

Not sure if this is an operator error or how the plugin exposes ports via docker but the general issue I did not need to do this before the docker upgrade.

**Comment by chrmarti** [comment]
@razilevin Please check if you have any user settings related to remote ports set:
<img width="1185" alt="image" src="https://user-images.githubusercontent.com/9205389/214243672-4a973d2b-ad32-4dc8-87a7-3990225b96a7.png">
(For `Remote: Ports Attributes` you'll have to click on the link to check if the JSON has any specific value.)

**Comment by alexr00** [comment]
@chrmarti the statically forwarded port comes from Remote Containers and is only shown in the UI in the Ports view, nothing else is done with it. Would you expect this user to also need to use VS Code's port forwarding to manually forward the port too?

**Comment by chrmarti** [comment]
@alexr00 My understanding is that the automatic port forwarding stopped working and that's why @razilevin tried with `"appPort"`.

**Comment by alexr00** [comment]
@razilevin can you share your ports related settings?

**Comment by razilevin** [comment]
The `appPort` did not work and was me trying to set everything that contained with the word port.
![image](https://user-images.githubusercontent.com/1204724/214588304-853e39d8-ac6c-45f7-bf2d-385882af2ba5.png)

Please note these are my user settings might be different than my workspace settings.
```json
{
    "remote.portsAttributes": {    
        "443": {
            "protocol": "https"
        },
        "8443": {
            "protocol": "https"
        }
    }
}
```

Honestly I updated docker via APT and poof I now have to manually add the port to be exposed by remote containers. 

So the fix is to remove these lines?

**Comment by alexr00** [comment]
> Please note these are my user settings might be different than my workspace settings.

Can you also share your workspace settings?

Do you manually run the process that listens on port 3000?

**Comment by razilevin** [comment]
Yes the process is manually run via the terminal. I wonder if the community is not seeing this issue then it is something on my linux machine I broke. Not sure what

![image](https://user-images.githubusercontent.com/1204724/214922554-4445d0e4-3fb9-4bec-91b1-addeba4b1e72.png)

**Comment by alexr00** [comment]
@razilevin thank you for the additional info. This doesn't seem to be a wide spread issue in the community, but I have gotten a few reports of ports not being detected that seem to be related to the specific process being launched. We discover ports to forwarded by watching for processes that are launched on the remote system and there's something about how some processes run that causes them to be missed by our watching.

Could you try the following:
1. Connect to your remote.
1. command "Developer: Set Log level" -> "Trace"
1. Start your process that listens on a port and wait about 5 seconds
1. Share the Output from "Remote Extension Host", "Shared", and "Window".

These logs will let me confirm that you are indeed hitting the process discovery issue.
One other thing you can try: set `"remote.autoForwardPortsSource": "output"`. "output" will not discover as many ports as "process" will, but it could unblock you.
